autocertkit/network_tests.py

```diff
@@ -1002,6 +1002,85 @@ def test_ping(self, session):
         return self._run_test(session)
 
 
+class MulticastTest(IperfTest):
+    """ Subclass that adds multicast specific operations"""
+    MULTICAST_IP = '226.94.1.1'
+    iperf_report_path = '/tmp/iperf.log'
+
+    def run_mulitcast_server(self):
+        """Start the iPerf server listening on a VM"""
+        log.debug('Starting multicast server')
+        self.server_ip = self.get_server_ip('eth0')
+        cmd_str = 'nohup iperf -y csv -s -u -B %s </dev/null &>%s &' \
+            % (self.MULTICAST_IP, self.iperf_report_path)
+        ssh_command(self.server_ip, self.username, self.password,
+                    cmd_str)
+
+    def run_multicast_client(self):
+        """Run test iperf command on droid VM"""
+        log.debug('Starting IPerf client')
+        cmd_str = 'iperf -c %s -u' % self.MULTICAST_IP
+        ssh_command(self.get_client_ip(), self.username, self.password,
+                    cmd_str)
+
+    def validate_multicast(self):
+        cmd_str = 'cat %s' % self.iperf_report_path
+        cmd_result = ssh_command(self.server_ip, self.username,
+                                 self.password, cmd_str)
+        log.debug('Iperf log on multicast server: %s' % cmd_result)
+        if self.MULTICAST_IP not in cmd_result:
+            raise TestCaseError('Error: Multicast test failed. iperf data: %s'
+                                 % cmd_result)
+
+        self.iperf_data = self.parse_iperf_line(cmd_result)
+
+    def run(self):
+        """This classes run test function"""
+        self.deploy_iperf()
+        self.run_mulitcast_server()
+        log.debug('IPerf deployed and multicast server started')
+        self.run_multicast_client()
+
+        iperf_data = self.validate_multicast()
+        return iperf_data
+
+
+class MulticastTestClass(IperfTestClass):
```

Comment:
@bpwook 
From Rob dobson:
If we're going to add a new feature to the product, we need to consider whether we should and can add extra tests for certification purposes. Given we're working on the multicast REQ right now I suggest that @liangd syncs up with @vivekkumac to discuss what sort of tests could be usefully added.
Adding a test in the ACK will ensure that hardware certified on our HCL supports multicast properly - it would give us the option to have it listed as a feature that any given NIC may or may not support.